query/src/org/labkey/query/reports/ReportsController.java

```diff
@@ -3187,11 +3187,11 @@ private JSONObject convertTree(CategoryHelper helper) // ViewCategory vc, Map<In
             ArrayList<JSONObject> views = new ArrayList<>();
             for (DataViewInfo view : helper.views)
             {
-                JSONObject viewJson = DataViewService.get().toJSON(getContainer(), getUser(), view);
+                JSONObject viewJson = DataViewService.get().toJSON(getUser(), view);
                 viewJson.put("name", view.getName());
                 viewJson.put("leaf", true);
                 viewJson.put("icon", view.getIconUrl().getLocalURIString());
-                viewJson.put("iconCls", view.getIconCls());
+                viewJson.put("iconCls", view.getIconCls() == null ? JSONObject.NULL : view.getIconCls());
```

Comment:
@labkey-jeckels For future reference, I've been using `JSONObject.wrap` for these sort of fixes.
```suggestion
                viewJson.put("iconCls", JSONObject.wrap(view.getIconCls()));
```

Reply:
@labkey-tchad thanks - I'll try to use that in the future